File: 백준/Silver/15650. N과 M （2）/N과 M （2）.cc
#include<iostream>
using namespace std;
void dfs(int n, int m, int start, int cnt, int arr[9]) {
	if (cnt == m) {
		for (int i = 0; i < m; i++) {
			cout << arr[i] << " ";
		}
		cout << "\n";
		return;
	}

	for (int i = start; i < n + 1;i++) {
		arr[cnt] = i;
		dfs(n,m,i + 1, cnt + 1, arr);
	}
}

int main() {
	int n; int m;
	cin >> n>>m;
	int arr[9];
	dfs(n,m,1, 0,arr);

	return 0;
}

File: 백준/Silver/15650. N과 M （2）/N과 M （2）.py
n, m = map(int, input().split())

arr = [0] * m  # Initialize arr with length m

def dfs(start, cnt):
    if cnt == m:
        for i in range(m):
            print(str(arr[i]) + " ", end="")
        print()
        return
    
    for i in range(start, n + 1):
        arr[cnt] = i
        dfs(i + 1, cnt + 1)  # i+1부터 탐색을 시작하여 오름차순을 보장

dfs(1, 0)
